Sources/SwiftSyntaxMacrosTestSupport/Assertions.swift

```diff
@@ -34,41 +34,76 @@ public struct NoteSpec {
   ///
   /// - Parameters:
   ///   - message: The expected message of the note
   ///   - line: The line to which the note is expected to point
   ///   - column: The column to which the note is expected to point
-  ///   - originatorFile: The file at which this ``NoteSpec`` was created, so that assertion failures can be reported at its location.
+  ///   - originatorFileID: The file ID at which this ``NoteSpec`` was created, so that assertion failures can be reported at its location.
+  ///   - originatorFilePath: The file path at which this ``NoteSpec`` was created, so that assertion failures can be reported at its location.
   ///   - originatorLine: The line at which this ``NoteSpec`` was created, so that assertion failures can be reported at its location.
+  ///   - originatorColumn: The column at which this ``NoteSpec`` was created, so that assertion failures can be reported at its location.
   public init(
     message: String,
     line: Int,
     column: Int,
-    originatorFile: StaticString = #file,
-    originatorLine: UInt = #line
+    originatorFileID: String = #fileID,
+    originatorFilePath: StaticString = #filePath,
+    originatorLine: UInt = #line,
+    originatorColumn: UInt = #column
   ) {
     self.message = message
     self.line = line
     self.column = column
-    self.originatorFile = originatorFile
+    self.originatorFileID = originatorFileID
+    self.originatorFilePath = originatorFilePath
     self.originatorLine = originatorLine
+    self.originatorColumn = originatorColumn
   }
 }
 
 func assertNote(
   _ note: Note,
   in expansionContext: BasicMacroExpansionContext,
-  expected spec: NoteSpec
+  expected spec: NoteSpec,
+  failingWith failureHandler: ((_ message: String, _ fileID: String, _ filePath: StaticString, _ line: UInt, _ column: UInt) -> Void)? = nil
 ) {
-  assertStringsEqualWithDiff(note.message, spec.message, "message of note does not match", file: spec.originatorFile, line: spec.originatorLine)
+  let failureHandler = failureHandler ?? defaultFailureHandler
+  if note.message != spec.message {
+    let message = describeDifferenceBetweenStrings(note.message, spec.message, "message of note does not match")
+    failureHandler(
+      message,
+      spec.originatorFileID,
+      spec.originatorFilePath,
+      spec.originatorLine,
+      spec.originatorColumn
+    )
+  }
   let location = expansionContext.location(for: note.position, anchoredAt: note.node, fileName: "")
-  XCTAssertEqual(location.line, spec.line, "line of note does not match", file: spec.originatorFile, line: spec.originatorLine)
-  XCTAssertEqual(location.column, spec.column, "column of note does not match", file: spec.originatorFile, line: spec.originatorLine)
+  if location.line != spec.line {
+    failureHandler(
+      "line \(location.line) of note does not match \(spec.line)",
+      spec.originatorFileID,
+      spec.originatorFilePath,
+      spec.originatorLine,
+      spec.originatorColumn
+    )
+  }
```

Comment:
Similar here, what about having `assertEqual` that takes a `failureHandler`?

(same in the other locations where you replaced `XCTAssertEqual`)